docs/llm_tools_invocation.md

```diff
@@ -0,0 +1,90 @@
+---
+title: "LLM Tools Invocation"
+---
+
+If you are developing an LLM integration and want to add support of LLM Tools invocation, then here are the steps to do that.
+
+## Step 0. Register the `intent` component.
+
+The intents are also automatically registered as LLM tools. However in order to do so, the `intent` component must be loaded. It does not happen automatically, so it is recommended to make sure it is loaded before using the tools for the first time.
+
+```python
+from homeassistant import setup
+...
+        if "intent" not in self.hass.config.components:
+            await setup.async_setup_component(self.hass, "intent", {})
+```
+
+## Step 1. Get available tools
+
+You can get the currently registered tools with `homeassistant.helpers.llm.async_get_tools(hass)`:
+
+```python
+from homeassistant.helpers import llm
+...
+        tools = list(llm.async_get_tools(self.hass))
+```
+
+## Step 2. Format the tool schema
+
+You need to pass the list of available tools to the LLM. Refer to the LLM API for the correct format to do that. You would probably need the tool specification in OpenAPI-compatible format. To get that, you can use `voluptuous_openapi.convert` function (don't forget to mention the dependency in your `manifest.json` file). Simplest example:
+
+```python
+from voluptuous_openapi import convert
+from homeassistant.helpers import llm
+...
+        tools = [
+            {
+                "name": tool.name,
+                "description": tool.description,
+                "parameters": convert(tool.parameters),
+            }
+            for tool in llm.async_get_tools(self.hass)
+        ]
+```
+
+You may need some additional processing.
+
+Pass the list to your LLM according to the API specification.
+
+## Step 3. Call the tool
+
+When the LLM response indicates it wants to call a tool, you need to prepare the `llm.ToolInput` object, then execute the `llm.async_call_tool` function.
+
+Example:
+
+```python
+from homeassistant.helpers import llm
+...
```

Comment:
Perhaps this example should be inline in a https://developers.home-assistant.io/docs/core/entity/conversation?_highlight=conversation#process example?